app/components/Blockchain/Transaction/Transaction.jsx

```diff
@@ -34,17 +39,45 @@ export default class Transaction extends React.Component<Props> {
         </div>
         <div className={styles.txLabelContainer}>{label}</div>
         <div className={styles.txAmountContainer}>{amount}</div>
-        <div className={styles.txToContainer}>{to}</div>
-        <span
-          className={classNames(styles.txidLink, className)}
+        <div className={styles.txToContainer}>
+          {to}
+          {to !== 'NETWORK FEES' && (
+            <CopyToClipboard
+              className={styles.copy}
+              text={to}
+              tooltip="Copy Public Address"
+            />
+          )}
+        </div>
+
+        {to === 'NETWORK FEES' ? (
+          <div className={styles.historyButtonPlaceholder} />
+        ) : (
+          <Button
+            className={styles.transactionHistoryButton}
+            renderIcon={ContactsAdd}
+            onClick={this.displayModal}
+          >
+            Add
+          </Button>
+        )}
+        <Button
+          className={styles.transactionHistoryButton}
+          renderIcon={InfoIcon}
           onClick={this.handleClick}
         >
-          {txid.substring(0, 32)}
-        </span>
+          View
+        </Button>
       </div>
     )
   }
 
+  displayModal = () => {
+    const { showAddContactModal, tx } = this.props
```

Comment:
you can do this all in one line const { showAddContactModal, tx: { to } } = this.props

Reply:
👍